# Exploración inicial: delitos de hurto por modalidades

**Archivo:** `Reporte_Hurto_por_Modalidades_Policía_Nacional_20260916.csv`

**Pregunta inicial:** 
¿Es posible predecir la cantidad de hurtos de motocicletas que se presentarán en un municipio durante el siguiente mes, utilizando el comportamiento histórico de los hurtos y las variables disponibles en el dataset?

In [2]:
import pandas as pd

df = pd.read_csv(
    "../datos/Reporte_Hurto_por_Modalidades_Policía_Nacional_20260916.csv"
)

df.head()


,DEPARTAMENTO,MUNICIPIO,CODIGO DANE,ARMAS MEDIOS,FECHA HECHO,GENERO,GRUPO ETARIO,TIPO DE HURTO,CANTIDAD
0,ATLÁNTICO,Barranquilla (CT),8001000,NO REPORTADO,05/03/2014,FEMENINO,ADULTOS,ARTÍCULO 239. HURTO AUTOMOTORES,1
1,META,Villavicencio (CT),50001000,NO REPORTADO,14/02/2014,FEMENINO,ADULTOS,ARTÍCULO 239. HURTO AUTOMOTORES,1
2,NORTE DE SANTANDER,Los Patios,54405000,NO REPORTADO,21/02/2014,MASCULINO,ADULTOS,ARTÍCULO 239. HURTO MOTOCICLETAS,1
3,ANTIOQUIA,Amagá,5030000,NO REPORTADO,03/01/2014,MASCULINO,ADULTOS,ARTÍCULO 239. HURTO MOTOCICLETAS,1
4,META,Villavicencio (CT),50001000,NO REPORTADO,22/02/2014,MASCULINO,ADULTOS,ARTÍCULO 239. HURTO AUTOMOTORES,1


In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 656635 entries, 0 to 656634
Data columns (total 9 columns):
 #   Column         Non-Null Count   Dtype 
---  ------         --------------   ----- 
 0   DEPARTAMENTO   656635 non-null  object
 1   MUNICIPIO      656635 non-null  object
 2   CODIGO DANE    656635 non-null  int64 
 3   ARMAS MEDIOS   656635 non-null  object
 4   FECHA HECHO    656635 non-null  object
 5   GENERO         655923 non-null  object
 6   GRUPO ETARIO   656430 non-null  object
 7   TIPO DE HURTO  656635 non-null  object
 8   CANTIDAD       656635 non-null  int64 
dtypes: int64(2), object(7)
memory usage: 45.1+ MB


In [7]:
print(f'Filas: {df.shape[0]:,}')
print(f'Columnas: {df.shape[1]}')

print('Nombres de las columnas:')
for numero, columna in enumerate(df.columns, start=1):
    print(f'{numero}. {columna}')



Filas: 656,635
Columnas: 9
Nombres de las columnas:
1. DEPARTAMENTO
2. MUNICIPIO
3. CODIGO DANE
4. ARMAS MEDIOS
5. FECHA HECHO
6. GENERO
7. GRUPO ETARIO
8. TIPO DE HURTO
9. CANTIDAD


In [8]:
df.isnull().sum()

DEPARTAMENTO       0
MUNICIPIO          0
CODIGO DANE        0
ARMAS MEDIOS       0
FECHA HECHO        0
GENERO           712
GRUPO ETARIO     205
TIPO DE HURTO      0
CANTIDAD           0
dtype: int64

In [9]:
df["CANTIDAD"].value_counts()
df["CANTIDAD"].describe()

count    656635.0
mean          1.0
std           0.0
min           1.0
25%           1.0
50%           1.0
75%           1.0
max           1.0
Name: CANTIDAD, dtype: float64

In [10]:
df["TIPO DE HURTO"].unique()

array(['ARTÍCULO 239. HURTO AUTOMOTORES',
       'ARTÍCULO 239. HURTO MOTOCICLETAS'], dtype=object)

In [11]:
# Separar el texto de TIPO DE HURTO en sus partes
df["ARTICULO_PENAL"] = df["TIPO DE HURTO"].str.extract(r"(ARTÍCULO \d+)")
df["TIPO_VEHICULO"] = df["TIPO DE HURTO"].str.extract(r"HURTO (\w+)$")

print(df["ARTICULO_PENAL"].unique())
print(df["TIPO_VEHICULO"].unique())

['ARTÍCULO 239']
['AUTOMOTORES' 'MOTOCICLETAS']


## Resultado inicial

- El dataset contiene información histórica sobre casos de hurto registrados por la Policía Nacional, reportando 656,635 datos.
- Los registros contienen información geográfica, temporal y características relacionadas con los hechos.
- La variable de fecha permitirá organizar los registros cronológicamente.
